### PR TITLE
sql: include schema prefix in create statements

### PR DESCRIPTION
**Pull request by rohany**
This PR changes the `crdb_internal.create_statements` table to include
schema qualifications in its create statements. This is needed for #50874,
as the create statements that dump uses are taken from this source.

Release note (sql change): The `SHOW CREATE` statement now contains the
schema of a table in the create statement.

### CODE REVIEW

File: pkg/sql/show_create_clauses.go [hunk 1]
```diff
@@ -169,16 +169,14 @@ func showForeignKeyConstraint(
 		if err != nil {
 			return err
 		}
-		fkDb, err := lCtx.getDatabaseByID(fkTable.ParentID)
+		fkTableName, err = getTableAsTableName(lCtx, fkTable, dbPrefix)
```
Comment (otan) [inline pkg/sql/show_create_clauses.go:172]: does this make more sense to be a method on `TableDescriptor`?

File: pkg/sql/pg_catalog.go [hunk 2]
```diff
@@ -1008,6 +1009,20 @@ func (r oneAtATimeSchemaResolver) getTableByID(id sqlbase.ID) (*TableDescriptor,
 	return table.Desc.TableDesc(), nil
 }
 
+func (r oneAtATimeSchemaResolver) getSchemaByID(
+	id sqlbase.ID,
+) (*sqlbase.ImmutableSchemaDescriptor, error) {
+	// TODO (rohany): This should use the descs.Collection.
```
Comment (otan) [inline pkg/sql/pg_catalog.go:1015]: what happens if we don't at this point in time? we don't have caching? is there correctness issues with doing it this way atm?

**Review by otan — COMMENTED** [review]
some small questions

File: pkg/sql/show_create_clauses.go (hunk 1 above)
Comment (rohany) [reply pkg/sql/show_create_clauses.go:172]: I'm not sure, I don't want to move this resolver interface into sqlbase. The functions seem somewhat specific to this process of displaying create statements, so I'm tempted to keep it here.

File: pkg/sql/pg_catalog.go (hunk 2 above)
Comment (rohany) [reply pkg/sql/pg_catalog.go:1015]: Nothing bad goes wrong, we'll just be doing a kv op every time. Work is in progress right now to add the appropriate API's to the descs.Collection.

File: pkg/sql/show_create_clauses.go (hunk 1 above)
Comment (otan) [reply pkg/sql/show_create_clauses.go:172]: minor nit: can we rename it such that `TableDescriptor` is in the name?

**Review by otan — APPROVED** [review]
LGTM

File: pkg/sql/pg_catalog.go (hunk 2 above)
Comment (otan) [reply pkg/sql/pg_catalog.go:1015]: And so uncommitted schemas and the like show up?
(or should it not show up)

Comment (rohany) [reply pkg/sql/pg_catalog.go:1015]: Yeah, they'll show up (which is expected during a transaction)

**Review by pbardea — APPROVED** [review]
Bulk IO changes LGTM

File: pkg/sql/show_create_clauses.go (hunk 1 above)
Comment (rohany) [reply pkg/sql/show_create_clauses.go:172]: done

**Pull request merged**